elasticsearch/connection/base.py

```diff
@@ -38,20 +47,54 @@ def __init__(
         use_ssl=False,
         url_prefix="",
         timeout=10,
+        headers=None,
+        http_compress=False,
+        cloud_id=None,
+        api_key=None,
         **kwargs
     ):
-        """
-        :arg host: hostname of the node (default: localhost)
-        :arg port: port to use (integer, default: 9200)
-        :arg url_prefix: optional url prefix for elasticsearch
-        :arg timeout: default timeout in seconds (float, default: 10)
-        """
+
+        if cloud_id:
+            try:
+                _, cloud_id = cloud_id.split(":")
+                parent_dn, es_uuid, _ = (
+                    binascii.a2b_base64(cloud_id.encode("utf-8")).decode("utf-8").split("$")
```

Comment:
we are using the `base64` wrapper in other places, maybe worth unifying on the low level `binascii`?